filter_plugins/opg_custom.py

```diff
@@ -2,27 +2,79 @@
 
 
 def dict_to_list(dict_to_convert):
+    """
+    :param dict_to_convert:
+    :return:
+    """
     if isinstance(dict_to_convert, dict):
         return [{k: v} for k, v in dict_to_convert.items()]
     else:
         raise errors.AnsibleFilterError('Supplied argument must be of type dict')
 
 
+def list_intersect(first, second):
+    """
+    Returns elements found in first that are in second
+    :param first:
+    :param second:
+    :return:
+    """
+    second = set(second)
+    return [item for item in first if item in second]
+
+
+def merge_custom_app_data(src=[], destination=[]):
+    """
+    :param src:
+    :param destination:
+    :return []:
+    Merges two lists
+    """
+
+    src_names = dest_names = []
+
+    for elem in src:
+        if 'name' in elem:
+            src_names.append(elem['name'])
+
+    for elem in destination:
+        if 'name' in elem:
+            dest_names.append(elem['name'])
+
+    intersects = list_intersect(src_names, dest_names)
+
+    try:
+        if intersects:
+            raise ValueError('Keys {} are redefined in target list'.format(intersects))
```

Comment:
Caught below and re-raised as ansible filter error in case we get another kind of exception